src/libAtomVM/bitstring.h

```diff
@@ -37,13 +37,21 @@ extern "C" {
         ( (((uint8_t *)(ptr))[1] << 8) | ((uint8_t *)(ptr))[0] )
 
     #define WRITE_16LE_UNALIGNED(t, ptr, val) \
-        *((t *) (ptr)) = ( (((uint8_t *)(&val))[1] << 8) | ((uint8_t *)(&val))[0] )
+        { \
+            ((uint8_t *)(ptr))[0] = ((uint16_t) val) & 0xff; \
+            ((uint8_t *)(ptr))[1] = (((uint16_t) val) >> 8) & 0xff; \
+        }
```

Comment:
I believe GCC compiles this anyway, but as far as I know the correct approach is: `do { ... } while (0)`. I don't know why.